libmogwai-schedule/schedule-service.c

```diff
@@ -1472,6 +1544,156 @@ schedule_cb (GObject      *obj,
     }
 }
 
+typedef struct
+{
+  MwsScheduleService *schedule_service;  /* (owned) */
+  GDBusMethodInvocation *invocation;  /* (owned) */
+} HoldData;
+
+static void
+hold_data_free (HoldData *data)
+{
+  g_clear_object (&data->invocation);
+  g_clear_object (&data->schedule_service);
+  g_free (data);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (HoldData, hold_data_free)
+
+static HoldData *
+hold_data_new (MwsScheduleService    *schedule_service,
+               GDBusMethodInvocation *invocation)
+{
+  g_autoptr(HoldData) data = g_new0 (HoldData, 1);
+  data->schedule_service = g_object_ref (schedule_service);
+  data->invocation = g_object_ref (invocation);
+  return g_steal_pointer (&data);
+}
+
+static void hold_cb (GObject      *obj,
+                     GAsyncResult *result,
+                     gpointer      user_data);
+
+static void
+mws_schedule_service_scheduler_hold (MwsScheduleService    *self,
+                                     GDBusConnection       *connection,
+                                     const gchar           *sender,
+                                     GVariant              *parameters,
+                                     GDBusMethodInvocation *invocation)
+{
+  /* Load the peer’s credentials so we can watch to see if it disappears in future. */
+  mws_peer_manager_ensure_peer_credentials_async (mws_scheduler_get_peer_manager (self->scheduler),
+                                                  sender, self->cancellable,
+                                                  hold_cb, hold_data_new (self, invocation));
+}
+
+static void
+hold_cb (GObject      *obj,
+         GAsyncResult *result,
+         gpointer      user_data)
+{
+  MwsPeerManager *peer_manager = MWS_PEER_MANAGER (obj);
+  g_autoptr(HoldData) data = user_data;
+  MwsScheduleService *self = data->schedule_service;
+  GDBusMethodInvocation *invocation = data->invocation;
+  g_autoptr(GError) local_error = NULL;
+
+  /* Finish looking up the sender. */
+  g_autofree gchar *sender_path = NULL;
+  sender_path = mws_peer_manager_ensure_peer_credentials_finish (peer_manager,
```

Comment:
I'd just break the line after the `=`...

Reply:
Meh, results in a slightly longer line.